From 0af45afd5e4789e40255bbdcaf4ac2347be71936 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Sat, 23 Mar 2019 23:42:03 +0100
Subject: [PATCH] [addons] add settings support for property condition (basic
 conditions only)

Before was it now possible to get a for Kodi defined condition on addon
settings. With this are the basic conditions also available for addon,
to define OS related parts.

The more complex conditions are not added, not sure if this goes then
to far to allow addons also this.
---
 xbmc/addons/settings/AddonSettings.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/xbmc/addons/settings/AddonSettings.cpp b/xbmc/addons/settings/AddonSettings.cpp
index 4fa4445e3d15..399218b4b321 100644
--- a/xbmc/addons/settings/AddonSettings.cpp
+++ b/xbmc/addons/settings/AddonSettings.cpp
@@ -22,6 +22,7 @@
 #include "guilib/LocalizeStrings.h"
 #include "messaging/ApplicationMessenger.h"
 #include "settings/SettingAddon.h"
+#include "settings/SettingConditions.h"
 #include "settings/SettingControl.h"
 #include "settings/SettingDateTime.h"
 #include "settings/SettingPath.h"
@@ -391,6 +392,13 @@ void CAddonSettings::InitializeControls()
 
 void CAddonSettings::InitializeConditions()
 {
+  CSettingConditions::Initialize();
+
+  // add basic conditions
+  const std::set<std::string>& simpleConditions = CSettingConditions::GetSimpleConditions();
+  for (const auto& condition : simpleConditions)
+    GetSettingsManager()->AddCondition(condition);
+
   GetSettingsManager()->AddDynamicCondition("InfoBool", InfoBool);
 }
 
